src/core_modules/capture-core/metaDataStoreLoaders/baseLoader/confirmLoadingSequence.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import { getContext } from '../context';
+import type { StalePrograms } from '../storeLoaders.types';
+
+export const confirmLoadingSequence = async (stalePrograms: StalePrograms) => {
+    const { storageController, storeNames } = getContext();
+    await Promise.all(
+        stalePrograms.map(async ({ id, version }) => {
+            const program = await storageController.get(storeNames.PROGRAMS, id);
+            const updatedProgram = {
+                ...program,
+                version,
+            };
+            await storageController.set(storeNames.PROGRAMS, updatedProgram);
```

Comment:
I have one question: do you know at which point in the process of initialising the cache the problem occurred? My feeling after reading this PR is that it happens during the transaction of javascript objects to entries in indexed DB. This line initiates such a transaction, and the shape of the statement suggests there's a chance the whole entry might be reinitialised.

Reply:
Basically there are multiple transactions taking place if the program version has changed and this fix ensures that the confirmation happens after all the transactions instead of confirming in the first transaction (by updating the version in the cache). I think what you describe here should be handled by the IndexedDB. We can have a chat if it is unclear.